Jenkinsfile.deploy

```diff
@@ -56,6 +57,9 @@ pipeline {
         // Bot author information. Set as params for easy testing.
         string(name: 'GIT_AUTHOR_BOT', defaultValue: 'bsig-gh-bot', description: 'From which author should the PR be created ?')
         string(name: 'BOT_CREDENTIALS_ID', defaultValue: 'bsig-gh-bot', description: 'Credentials for PR creation')
+
+        // PR parameter to be filled out for PR checks
+        string(name: 'PR_TARGET_BRANCH', defaultValue: '', description: 'What is the target branch of the PR?')
```

Comment:
What is the purpose of this parameter? This `Jenkinsfile.deploy` is not used in PRs, it's used in the nightly and release pipelines.

Reply:
here we do want to reuse the nightly pipelin flow for BDD tests from runtimes testing as it should do the exact same deployment part, but on a specific repository.

The problem here is that `optaplanner` repository is handling deploy/promote for all optaplanner repositories (optaplanner, optaweb[...],optaplanner-quickstarts)

On a PR check, we have a `source` branch and a `target` branch. It first checks if similar source branch exists and if not, it fallbacks to target branch.
That could happen if a change happens on both kogito-runtimes and optaplanner, so with same branch name (but not on optaweb[...])
So , in `GIT_BRANCH` param we sent the source branch. but if that branch does not exist on optaweb[...] repos, it will fail and so we need the target branch to checkout correct version (master or 8.1.x or ... ), hence the `PR_TARGET_BRANCH`  parameter ...

In an ideal world, all optaplanner repositories would have their own deploy/promote specific jenkinsfile/jobs and so we could just do the check if source branch is existing before calling the job. We can do that on runtimes/apps/examples/images/operator as each job is specific to the repository itself.

Else we need to copy that deploy job and create a new one but it would be exactly the same Jenkinsfile apart from that parameter and the correct checkouts. and so could be harder to maintain ...

Reply:
@rsynek is there anything we should change or do you agree with that ?

Reply:
I see, thanks for the clarification @radtriste .